app/src/main/java/de/danoeh/antennapod/dialog/TagSettingsDialog.java

```diff
@@ -27,28 +27,49 @@
 import io.reactivex.schedulers.Schedulers;
 
 import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
 
 public class TagSettingsDialog extends DialogFragment {
     public static final String TAG = "TagSettingsDialog";
     private static final String ARG_FEED_PREFERENCES = "feed_preferences";
+    private static final String ARG_MULTI_FEED_PREFERENCES = "multi_feed_preferences";
     private List<String> displayedTags;
     private EditTagsDialogBinding viewBinding;
     private TagSelectionAdapter adapter;
 
-    public static TagSettingsDialog newInstance(FeedPreferences preferences) {
+    private static TagSettingsDialog newInstance() {
         TagSettingsDialog fragment = new TagSettingsDialog();
         Bundle args = new Bundle();
-        args.putSerializable(ARG_FEED_PREFERENCES, preferences);
+        args.putSerializable(ARG_FEED_PREFERENCES, null);
         fragment.setArguments(args);
         return fragment;
     }
 
+    public static TagSettingsDialog newInstance(FeedPreferences preferences) {
+        TagSettingsDialog fragment = newInstance();
+        fragment.getArguments().putSerializable(ARG_FEED_PREFERENCES, preferences);
+        return fragment;
+    }
+
+    public static TagSettingsDialog newInstance(ArrayList<FeedPreferences> preferencesList) {
+        TagSettingsDialog fragment = newInstance();
+        fragment.getArguments().putSerializable(ARG_MULTI_FEED_PREFERENCES, preferencesList);
+        return fragment;
+    }
+
     @NonNull
     @Override
     public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
-        FeedPreferences preferences = (FeedPreferences) getArguments().getSerializable(ARG_FEED_PREFERENCES);
-        displayedTags = new ArrayList<>(preferences.getTags());
+        List<FeedPreferences> feedPreferencesList = getFeedPreferences();
+        Set<String> commonTags = new HashSet<>(feedPreferencesList.get(0).getTags());
+
+        for (FeedPreferences preference : feedPreferencesList) {
+            commonTags.retainAll(preference.getTags());
+        }
+        displayedTags = new ArrayList<>(commonTags);
```

Comment:
How about showing a small warning text when some tags are not shown because they are not used in all feeds? Then users don't wonder where their tags went.

Alternatively, it would be pretty cool to still show tags that are not used in all feeds but with some transparency. Probably makes the dialog a bit more complex but I think it would result in the best user experience.

Reply:
> How about showing a small warning text when some tags are not shown because they are not used in all feeds? 

Added